platforms/common/include/px4_platform_common/board_common.h

```diff
@@ -355,6 +355,22 @@ typedef enum PX4_SOC_ARCH_ID_t {
 
 } PX4_SOC_ARCH_ID_t;
 
+#define PX4_SOC_ARCH_ID_UNUSED         0x0000
```

Comment:
added these defines because the compiler was not properly evaluating the #if macro at https://github.com/ericjohnson97/px4-firmware-modalai/blob/ericjohnson97/sitl-fixes/platforms/posix/src/px4/common/drv_hrt.cpp#L59

This is probably broken for voxl2 now since the voxl2 board config defines PX4_SOC_ARCH_ID_VOXL2 as an enum and I don't think enums are evaluated at compilation

https://github.com/ericjohnson97/px4-firmware-modalai/blob/ericjohnson97/sitl-fixes/boards/modalai/voxl2/src/board_config.h#L52C25-L52C46